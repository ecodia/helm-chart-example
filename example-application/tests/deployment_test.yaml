suite: test deployment
templates:
  - deployment.yaml
tests:
  - it: should be of type Deployment
    asserts:
      - isKind:
          of: Deployment
  - it: manifest should match snapshot
    set:
      image:
        repository: test-docker-registry
        tag: 1.2.3-TEST
    asserts:
      - matchSnapshot:
          path: spec
