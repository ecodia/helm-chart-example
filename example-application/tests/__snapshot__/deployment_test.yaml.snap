manifest should match snapshot:
  1: |
    replicas: 1
    selector:
      matchLabels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: example-application
    template:
      metadata:
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: example-application
      spec:
        containers:
        - image: test-docker-registry:1.2.3-TEST
          imagePullPolicy: IfNotPresent
          livenessProbe:
            httpGet:
              path: /
              port: http
          name: example-application
          ports:
          - containerPort: 80
            name: http
            protocol: TCP
          readinessProbe:
            httpGet:
              path: /
              port: http
          resources: {}
